### DISCUSSION TITLE
Issues with the HelmRelease Controller

### DISCUSSION THREAD

**Discussion by apeschel** [fluxcd/flux2#2605] [Q&A] [answered]
There's a couple of issues with HelmReleases that I've run into recently:

1. HelmRelease resources can't depend on Kustomization resources, or vice versa.
2. The HelmRelease controller doesn't seem to support SOPS decryption as the Kustomization controller does.

Has anyone else run into thiese issues? Are there workarounds for them?

**Comment by kingdonb** [comment, marked as answer]
> HelmRelease resources can't depend on Kustomization resources

But Kustomization resources can `wait` for HelmRelease resources that they apply (or health check resources that are expected to be created by the HelmReleases by name with `healthChecks`. This way you can build Flux Kustomizations that depend on HelmReleases, by encapsulating the HelmRelease in a Kustomization and setting up the dependency between the Kustomizations.

> The HelmRelease controller doesn't seem to support SOPS decryption

When every HelmRelease is applied from a Kustomization, the issue is mitigated because you can do your decryption in the parent Kustomization, and include the secret in the HelmRelease with a [values override](https://fluxcd.io/docs/components/helm/helmreleases/#values-overrides) which can take values from any configmap or secret placed adjacent to the HelmRelease.
